test/irmin-pack/common.ml

```diff
@@ -179,6 +179,33 @@ module Alcotest = struct
 
   let int63 = testable Int63.pp Int63.equal
 
+  let check_raises_pack_error msg pass f =
+    Lwt.catch
+      (fun x ->
+        let* _ = f x in
+        Alcotest.failf
+          "Fail %s: expected function to raise, but it returned instead." msg)
+      (function
+        | Irmin_pack_unix.Errors.Pack_error e as exn -> (
+            match pass e with
+            | true -> Lwt.return_unit
+            | false ->
+                Alcotest.failf
+                  "Fail %s: function raised unexpected exception %s" msg
+                  (Printexc.to_string exn))
+        | exn -> (
+            let exn_str = Printexc.to_string exn in
+            match Astring.String.is_prefix ~affix:"Alcotest" exn_str with
+            (* Unfortunately we cannot access Alcotest's internal [Check_error]
+               so we do a dirty filter on its string representation to re-raise
+               instead of triggering another [Alcotest.failf]. *)
+            | true -> raise exn
+            | false ->
+                Alcotest.failf
+                  "Fail %s: expected function to raise Pack_error, but it \
+                   raised %s instead"
+                  msg exn_str))
```

Comment:
How about something like this instead with `Lwt.try_bind`? So we don't have to catch alcotest's exceptions
```suggestion
Lwt.try_bind (f x)
  (fun _ -> Alcotest.failf "Fail %s: expected function to raise, but it returned instead." msg)
  (function
    | Irmin_pack_unix.Errors.Pack_error e as exn -> (
            match pass e with
            | true -> Lwt.return_unit
            | false ->
                Alcotest.failf
                  "Fail %s: function raised unexpected exception %s" msg
                  (Printexc.to_string exn))
    | exn -> Alcotest.failf "Fail %s: expected function to raise Pack_error, but it \
                   raised %s instead" msg (Printexc.to_string exn))
```